internal/trigger/users/users_rate.go

```diff
@@ -38,18 +40,11 @@ func Rate() api.Builder {
 }
 
 func NewWorker(concurrency int) api.WorkTriggerer {
-	return func(workTriggered chan<- bool, stop <-chan bool, workDone <-chan bool, _ options.RunOptions) {
-		for range concurrency {
-			workTriggered <- true
-		}
+	return func(ctx context.Context, workers *workers.PoolManager, _ options.RunOptions) {
+		pool := workers.NewPool(concurrency)
+		workerCtx := pool.Start(ctx)
 
-		for {
-			select {
-			case <-stop:
-				return
-			case <-workDone:
-				workTriggered <- true
-			}
+		for pool.Queue(workerCtx) {
```

Comment:
Could this end up in an infinate loop if the worker pool's max iterations is == 0?

Reply:
No. I don't think so, because the parent context (ctx) is passed in from here with a timeout:
https://github.com/form3tech-oss/f1/blob/034d1444102ccbab50994ebbe5106efb8131746e/internal/run/test_runner.go#L235